### PR TITLE
Support translation dicts in API object creation

### PR DESCRIPTION
**Pull request by rowanseymour**
Introduces `TranslatableField` which can be a simple string or a translation dict. This is then used for `CampaignEvent.message` and `Broadcast.text`/`Broadcast.language_dict`

From discussion with @nicpottier, we agreed it makes sense to make `CampaignEvent.message` and `Broadcast.text` *always* stored as a dict, but these endpoints would still take a simple string - in the same way that Send Message dialog would still just take untranslated text. So planning to handle those changes in later PR.

### CODE REVIEW

**Comment by nicpottier** [comment]
Do we allow campaign event message updates? There's some trickiness there I think since those are flows under the hood, and we MUST have a value in the dictionary for the flow's base language otherwise all sorts of bad things will happen.

**Comment by nicpottier** [comment]
@ericnewcomer should look at this too but I think if we move Broadcasts to have only a dict then we need to introduce a base_language to Broadcast as well. We need to know what the "default" language was/is for all these objects, and when we just have a dict we don't know it by default. It works on CampaignEvents because they are always tied to a flow which has a baselanguage, but Broadcast doesn't have that to lean on.

File: temba/api/v2/serializers.py [hunk 1]
```diff
@@ -133,8 +132,13 @@ def save(self):
             contact_urn = contact.urn_objects[urn]
             recipients.append(contact_urn)
 
+        # TODO remove Broadcast.text
+        text = next(iter(self.validated_data['text'].values()))
```
Comment (norkans7) [inline temba/api/v2/serializers.py]: what about choosing the text for the org primary language for text?

**Comment by rowanseymour** [comment]
@nicpottier Broadcast already has base_language field so maybe need to check how we're using (also just to be clear this PR doesn't fundamentally change broadcasts)

**Comment by rowanseymour** [comment]
@ericnewcomer not quite following - you can't create a media using broadcast over the API

**Comment by ericnewcomer** [comment]
Exactly. This endpoint should let you set the media in the same way.

**Pull request merged**